lib/css/components/sidebar-widgets.less

```diff
@@ -0,0 +1,245 @@
+        tr.s-sidebarwidget--item {
+            display: table-row;
+            td {
+                padding: 0;
+            }
+        }
+
+        padding: calc(var(--_sw-content-spacing-outer) - var(--_sw-content-spacing-inner)) calc(var(--_sw-content-px) - var(--_sw-content-spacing-inner));
+        border-collapse: separate;
+        border-spacing: var(--_sw-content-spacing-inner);
+    }
+
+    &:not(.s-anchors) a:not(.button):not(.s-btn):not(.post-tag):not(&--action):not(.s-user-card--link) {
+        &,
+        &:visited {
+            color: var(--black-500);
+        }
+    }
+
+    &:before { // [1]
+        content: '';
+        display: block;
+        margin-top: -1px;
+    }
+
+    &:after { // [2]
+        content: '';
+        position: absolute;
+        top: -1px;
+        right: -1px;
+        left: -1px;
+        // TODO: this makes no sense. revisit.
+        height: calc(2px + var(--br-sm)); // we need 2px border + 2px border radius to have the correct corner shape
+        border-radius: var(--br-sm);
+        pointer-events: none; // if the top item is clickable, then we don't want to prevent clicking the top 2 pixels
+        clip-path: polygon(0% 0%, 100% 0%, 100% 50%, 0% 50%); // [3]
+        border: 1px solid var(--_sw-bc);
+    }
+
+    position: relative; // so that it's the positioning parent for the :after
+    border: 1px solid var(--_sw-bc);
+    border-radius: var(--br-sm);
+    box-shadow: var(--bs-sm);
+    font-size: var(--fs-body1);
+    background-color: var(--white);
+
+    // COLOR ALTERNATIVES
+    .alternate-color(blue);
+    .alternate-color(yellow);
+    .alternate-color(green);
+}
+
+
+// COLOR ALTERNATIVES
+.alternate-color(@name) {
+    &.s-sidebarwidget__@{name} {
+        --_sw-bc: var(~"--@{name}-200");
+
+        .highcontrast-mode({
+            --_sw-bc: var(~"--@{name}-700");
+        });
+
+        border-color: var(--_sw-bc);
+        background-color: var(~"--@{name}-050");
+
+        .s-sidebarwidget--header {
+            background-color: var(~"--@{name}-100");
+            color: var(--fc-medium);
+        }
+
+        .s-sidebarwidget--header,
+        .s-sidebarwidget--content,
+        &:after {
+            border-color: var(--_sw-bc);
+        }
+    }
+}
+
+// [1] We must support things like collapsible (in particular invisble) elements, wrapper elements,
+// etc. Therefore every .-content and .-header must stand on its own; we cannot rely on things
+// like :first-child, because the *first* child may not be the first *visible* child, and it may
+// also be the :first-child of some wrapper. This is why every .-header and .-content has a
+// border-top. But because you shouldn't see the first visible item's or header's top border
+// (the widget itself provides for that border), we shift everything up by one pixel with the following pseudo-element.
+
+// [2] However, because of the above, the top item's divider line now sits above the .s-sidebarwidget's top
```

Comment:
Seeing as these comments no longer live in their original context, they could probably benefit from some slight rewording.

Reply:
Makes sense. Will address this.